client/state/data-layer/wpcom/read/streams/index.js

```diff
@@ -100,6 +101,16 @@ const streamApis = {
 		path: ( { streamKey } ) => `/read/feed/${ streamKeySuffix( streamKey ) }/posts`,
 		dateProperty: 'date',
 	},
+	discover: {
+		path: () => '/read/tags/cards',
+		dateProperty: 'date',
+		query: ( extras, { tags } ) =>
+			getQueryString( {
+				...extras,
+				tags: tags ? Object.values( tags )?.map( ( tag ) => tag.slug ) : [],
```

Comment:
Noting a couple things for us to circle back to later.

Here, if the request to retrieve tags has not completed we end up with an empty feed on discover that never updates. You can test this by reloading on 'Follow' and instantly hitting the 'Discover' tab before the followed tags have loaded, a list will never resolve on the discover tab.  With something like useQuery we would make the hook dependent on tags which would be one way to resolve it, there is probably another way to resolve this in this structure but Im not exactly sure how atm.. so pointing this out for later.

I imagine in the future we might will fill in some default tags to the empty array to ensure there is some discover feed even hen the user has no tags followed? If we do that this bug will evolve to showing the discover feed for those base tags in that circumstance, and never updating to the appropriate feed for the followed tags once those are retrieved.

Reply:
We have a list of default tags that we could possible use in cases where no tags are found. This would be a good/simple workaround until we improve the UX around onboarding users to tags.

Reply:
That will be great to prevent the blank feed. I think we will still need to circle back to ensure the feed updates when the users followed tags are received though. We don't have to do that here.